deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/resource/Environs.java

```diff
@@ -161,6 +162,11 @@ public Response create(
         String operator = sc.getUserPrincipal().getName();
         String envName = environBean.getEnv_name();
         List<EnvironBean> environBeans = environDAO.getByName(envName);
+        if (environBean.getStage_type() == EnvType.DEFAULT) {
+            throw new TeletaanInternalException(Response.Status.BAD_REQUEST,
+                    "DEFAULT stage type is not allowed! Please one of the following Stage Types: " +
+                        "PRODUCTION, CONTROL, CANARY, STAGING, LATEST, DEV.");
+        }
```

Comment:
Have you considered implementing this logic in. `environBean.validate();`?

Reply:
Or use validation support `ValidationMethod` as in https://github.com/pinterest/teletraan/blob/d78f4264ba9802a45e57089bed670a38a15149a8/deploy-service/common/src/main/java/com/pinterest/deployservice/bean/PromoteBean.java#L182

Reply:
Yeah, I can move the logic to `environBean.validate()` for continuity